src/connect-to-browser.ts

```diff
@@ -87,28 +87,35 @@ export const connectToBrowser = async (
 ) => {
   // I acknowledge that this code is gross and should be refactored
   // Constraints:
-  // - If there is no browser in the config, multiple concurrent processes should only start 1 new browser
-  // - If there is a killed browser in the config, multiple concurrent processes should only start 1 new browser
-  // - If there "starting" in the config but nothing is really starting, multiple concurrent processes should only start 1 new browser
+  // - If there is no browser in the cache, multiple concurrent processes should only start 1 new browser
+  // - If there is a killed browser in the cache, multiple concurrent processes should only start 1 new browser
+  // - If there "starting" in the cache but nothing is really starting, multiple concurrent processes should only start 1 new browser
   // TODO: Idea: use a state machine!!!
```

Comment:
This code is still pretty gross, but it has held together way better than I expected. I'm leaving this comment here because I would still like to come back and rewrite it, but for the time being it works and is pretty resilient and "robust enough".